### ISSUE TITLE
clist: non-deterministic test failure

### ISSUE THREAD

**Issue by ebuchman** [tendermint/tendermint#2227]
Happened in unrelated PR:
- https://github.com/tendermint/tendermint/pull/2205/
- https://circleci.com/gh/tendermint/tendermint/20143#tests/containers/3

```
--- FAIL: TestWaitChan (1.26s)
	clist_test.go:271: number of pushed items (99) not equal to number of seen items (98)
```

**Comment by ValarDragon** [comment]
Same error as #2206. Strangely enough, it was on the same PR as well. Though this PR is certainly unrelated

**Comment by ebuchman** [comment]
Happened again in https://circleci.com/gh/tendermint/tendermint/31639?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```
?   	github.com/tendermint/tendermint/libs/autofile/cmd	[no test files]
Pushed 0K elements so far...
stopped
stopped
stopped
Scanner 6 restartCounter: 2 counter: 1183
stopped
Scanner 8 restartCounter: 1 counter: 612
stopped
stopped
Scanner 5 restartCounter: 2 counter: 1109
Scanner 0 restartCounter: 2 counter: 1967
stopped
stopped
Scanner 7 restartCounter: 1 counter: 1003
Scanner 3 restartCounter: 1 counter: 1027
Scanner 1 restartCounter: 2 counter: 1556
stopped
stopped
Scanner 2 restartCounter: 2 counter: 1409
Scanner 9 restartCounter: 2 counter: 1275
Scanner 4 restartCounter: 2 counter: 1321
--- FAIL: TestWaitChan (1.33s)
	clist_test.go:284: number of pushed items (99) not equal to number of seen items (98)
FAIL
coverage: 64.6% of statements
FAIL	github.com/tendermint/tendermint/libs/clist	1.378s
Exited with code 1
```

**Comment by KrzysiekJ** [comment]
Apparently again in https://circleci.com/gh/tendermint/tendermint/40785?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link .

```
Pushed 0K elements so far...
stopped
Scanner 6 restartCounter: 1 counter: 419
stopped
Scanner 0 restartCounter: 1 counter: 757
stopped
stopped
Scanner 5 restartCounter: 1 counter: 455
stopped
Scanner 2 restartCounter: 1 counter: 566
stopped
stopped
Scanner 4 restartCounter: 1 counter: 411
stopped
Scanner 1 restartCounter: 1 counter: 615
stopped
Scanner 3 restartCounter: 1 counter: 474
Scanner 7 restartCounter: 1 counter: 342
stopped
Scanner 9 restartCounter: 1 counter: 392
Scanner 8 restartCounter: 1 counter: 388
--- FAIL: TestWaitChan (1.27s)
    clist_test.go:284: number of pushed items (99) not equal to number of seen items (97)
FAIL
coverage: 64.6% of statements
FAIL	github.com/tendermint/tendermint/libs/clist	1.304s
Exited with code 1
```

**Comment by kevlubkcm** [comment]
hopefully closed by #3575

**Comment by marbar3778** [comment]
we haven't observed this failure in a while. will close this issue but if it resurfaces i would recommend we debug a bit instead of commenting in this issue that we saw it.
